### PR TITLE
Bugfix: Transform NULL values for SF Bulk API, which expects "#N/A"

### PR DESCRIPTION
**Pull request by asedge**
Need to transform `nil` into `#N/A` for the Salesforce Bulk API to pick up the change. See [here](https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/datafiles_prepare_csv.htm#:~:text=Empty%20field%20values%20are%20ignored%20when%20you%20update%20records.%20To%20set%20a%20field%20value%20to%20null%2C%20use%20a%20field%20value%20of%20%23N/A.).

### CODE REVIEW

File: lib/active_force/bulk/records.rb [hunk 1]
```diff
@@ -16,10 +16,25 @@ def to_csv
...
-        data = records.map { |r| r.sort.pluck(-1) }
+        data = records.map do |r|
+           r.transform_values { |v| transform_value_for_sf(v) }.sort.pluck(-1)
+        end
         new(headers: headers, data: data)
       end
+
+      # SF expects a special value for setting a column to be NULL.
+      def self.transform_value_for_sf(value)
+        case value
```
Comment (swpreethi) [inline lib/active_force/bulk/records.rb:29]: 🥇 
Dp you wanna store the '#N/A' value in any constant ?

**Review by swpreethi — APPROVED** [review]
Glad you found this. Left a non-blocking comment

File: lib/active_force/bulk/records.rb (hunk 1 above)
Comment (asedge) [reply lib/active_force/bulk/records.rb:29]: Hadn't thought about it but it sounds like a good suggestion to me, thanks!

File: lib/active_force/bulk/records.rb [hunk 2]
```diff
@@ -16,10 +18,25 @@ def to_csv
...
+        when NilClass
+          NULL_VALUE
+        when Time
```
Comment (bfrey08) [inline lib/active_force/bulk/records.rb:34]: Does this need to be DateTime and Time? What about just Date.

Comment (bfrey08) [reply lib/active_force/bulk/records.rb:34]: After further conversation - it looks like DateTime is already handled through calling the method: 

`    def translate_to_sf attributes
      attrs = attributes.map do |attribute, value|
        field = fields[attribute.to_sym]
        [field.sfdc_name, field.value_for_hash(value)]
      end
      Hash[attrs]
    end

 `
 
 LGTM!

**Review by bfrey08 — APPROVED** [review]
LGTM!

**Pull request merged**